game-core/src/main/java/games/strategy/triplea/settings/SettingsWindow.java

```diff
@@ -181,26 +187,32 @@ private JComponent buildButtonPanel() {
   }
 
   private void saveSettings() {
-    getSelectedSettings().ifPresent(settings -> {
-      final SaveFunction.SaveResult saveResult = SaveFunction.saveSettings(settings, ClientSetting::flush);
+    getSelectedSelectionComponents().ifPresent(selectionComponents -> {
+      final SaveFunction.SaveResult saveResult = SaveFunction.saveSettings(selectionComponents, ClientSetting::flush);
       JOptionPane.showMessageDialog(dialog, saveResult.message, "Results", saveResult.dialogType);
     });
   }
 
   private void resetSettings() {
-    getSelectedSettings().ifPresent(settings -> settings.forEach(ClientSettingSwingUiBinding::reset));
+    getSelectedSelectionComponents()
+        .ifPresent(selectionComponents -> selectionComponents.forEach(SelectionComponent::reset));
   }
 
   private void resetSettingsToDefault() {
-    getSelectedSettings().ifPresent(settings -> settings.forEach(ClientSettingSwingUiBinding::resetToDefault));
+    getSelectedSelectionComponents()
+        .ifPresent(selectionComponents -> selectionComponents.forEach(SelectionComponent::resetToDefault));
   }
 
-  private Optional<List<ClientSettingSwingUiBinding>> getSelectedSettings() {
+  private Optional<List<SelectionComponent<JComponent>>> getSelectedSelectionComponents() {
     assert tabbedPane != null;
 
     final int selectedTabIndex = tabbedPane.getSelectedIndex();
-    return (selectedTabIndex != -1)
-        ? Optional.of(getSettingsByType(settingTypes.get(selectedTabIndex)))
-        : Optional.empty();
+    if (selectedTabIndex == -1) {
```

Comment:
I'm wondering when this would happen.
I get that `tabbedPane.getSelectedIndex()` will return -1 when there's no tab selected, but I don't think this should ever happen.
According to the javadocs of [`JTabbedPane#setSelectedIndex(int)`](https://docs.oracle.com/javase/10/docs/api/javax/swing/JTabbedPane.html#setSelectedIndex(int)):
> [...] If a -1 value is specified when the tabbedpane contains one or more tabs, then the results will be implementation defined.

Which would indicate that there may be implementations that allow -1 values for tabbed panes with non-0 amounts of tabs.
I case this really shouldn't happen, I'd suggest throwing an IllegalStateException here instead of returning an empty optional, which makes using an optional obsolete, and reduces the calling code's complexity

Reply:
I think the `-1` case is more typically realized when (accidentally?) calling `getSelectedIndex()` before any tabs have been added.  Maybe that's more important for `JTabbedPane`s that dynamically add and remove tabs?

So, I agree, that, since we're using a static set of tabs, and we create them early on, it would be a programmer error if the selected index was ever `-1`.  However, since it is a programmer error internal to a class, I think an assertion is more appropriate than an unchecked exception.  (And, even if assertions are disabled, an `IndexOutOfBoundsException` unchecked exception will be thrown anyway when passing `-1` to `List#get()`.)  82fed14de